java/code/src/com/redhat/rhn/taskomatic/task/ForwardRegistrationTask.java

```diff
@@ -75,20 +86,19 @@ public void execute(JobExecutionContext arg0) throws JobExecutionException {
                 log.debug("{} RegCacheItems found to forward", forwardRegistration.size());
                 List<SCCRegCacheItem> deregister = SCCCachingFactory.listDeregisterItems();
                 log.debug("{} RegCacheItems found to delete", deregister.size());
-                List<Credentials> credentials = CredentialsFactory.lookupSCCCredentials();
                 SCCConfig sccConfig = new SCCConfig(url, "", "", uuid);
                 SCCClient sccClient = new SCCWebClient(sccConfig);
                 SCCSystemRegistrationManager sccRegManager = new SCCSystemRegistrationManager(sccClient);
                 sccRegManager.deregister(deregister, false);
-                credentials.stream()
-                    .filter(Credentials::isPrimarySCCCredential)
-                    .findFirst()
-                    .ifPresent(primaryCredentials -> sccRegManager.register(forwardRegistration, primaryCredentials));
+                optPrimCred.ifPresent(primaryCredentials ->
```

Comment:
instead of calling ifPresent multiple times we could put all the logic that requires the credentials to be there in a helper function and take the credentials as parameter so we only have to have to call ifPresent one time for the helper function and the rest can use the credentials normal.